cmd/mafIndels/mafIndels.go

```diff
@@ -8,44 +8,57 @@ import (
 			//TODO: to improve clarity, consider using assembly_k first, and then setting assembly_del:=assembly_k if in fact the k line is a del line
 
+			//verify line 0 is indeed species_ins
+			if assembly_ins != species_ins {
+				log.Fatalf("species_ins was incorrect. Please check you have a pairwise maf file, and entered species_ins and species_del correctly") //otherwise fatal
+			}
+
 			//get eC species_del lines
 			if mafRecords[i].Species[k].ELine != nil {
-				if mafRecords[i].Species[k].ELine.Status == 'C' && assembly_del == species_del { //I decided to check for both Status and Src here because they are on the same data level
+				if mafRecords[i].Species[k].ELine.Status == 'C' && assembly_del == species_del && mafRecords[i].Species[0].SLine != nil { //I decided to check for both Status and Src here because they are on the same data level, as well as get corresponding s species_ins lines
 
-					//get corresponding s species_ins lines
-					if assembly_ins != species_ins { //verify line k-1 is indeed species_ins
-						log.Fatalf("species_ins was incorrect. Please check you have a pairwise maf file, and entered species_ins and species_del correctly") //otherwise fatal
-					}
-					if mafRecords[i].Species[k-1].SLine != nil { //if corresponding species_ins line is an s line
+					//convert maf to bed, continued
+					current_del := bed.Bed{Chrom: chrom_del, ChromStart: mafRecords[i].Species[k].ELine.Start, ChromEnd: mafRecords[i].Species[k].ELine.Start + mafRecords[i].Species[k].ELine.Size, Name: "del_eC", Score: int64(mafRecords[i].Score)} //get chrom,start,end,name,score
+					current_ins := bed.Bed{Chrom: chrom_ins, ChromStart: mafRecords[i].Species[0].SLine.Start, ChromEnd: mafRecords[i].Species[0].SLine.Start + mafRecords[i].Species[0].SLine.Size, Name: "ins_eC", Score: int64(mafRecords[i].Score)}
+					bedList_del = append(bedList_del, &current_del) //append to growing bed
+					bedList_ins = append(bedList_ins, &current_ins)
 
-						//convert maf to bed, continued
-						current_del := bed.Bed{Chrom: chrom_del, ChromStart: mafRecords[i].Species[k].ELine.Start, ChromEnd: mafRecords[i].Species[k].ELine.Start + mafRecords[i].Species[k].ELine.Size, Name: "del", Score: int64(mafRecords[i].Score)} //get chrom,start,end,name,score
-						current_ins := bed.Bed{Chrom: chrom_ins, ChromStart: mafRecords[i].Species[k-1].SLine.Start, ChromEnd: mafRecords[i].Species[k-1].SLine.Start + mafRecords[i].Species[k-1].SLine.Size, Name: "ins", Score: int64(mafRecords[i].Score)}
-						bedList_del = append(bedList_del, &current_del) //append to growing bed
-						bedList_ins = append(bedList_ins, &current_ins)
+					//get eI species_del lines
+				} else if mafRecords[i].Species[k].ELine.Status == 'I' && assembly_del == species_del && mafRecords[i].Species[0].SLine != nil {
+					if mafRecords[i].Species[0].SLine != nil { //if corresponding species_ins line is an s line
```

Comment:
I think you already have this check above.  Maybe you could add:
assembly_del == species_del && mafRecords[i].Species[0].SLine != nil
to the check on line 34 since you need to check this on both lines 35 and 44?